web/ui/src/Services/Service.js

```diff
@@ -141,9 +141,18 @@
         fetchInstances() {
             let deferred = $q.defer();
             resourcesFactory.v2.getServiceInstances(this.id)
-                .then(data => {
-                    // TODO - dont blow away existing instances
-                    this.instances = data.map(i => new Instance(i));
+                .then(results => {
+                    // console.log(`fetched ${data.length} instances for ${this.id}`);
+                    results.forEach(data => {
```

Comment:
Please comment here that you are merging instances to prevent anything bounce to the instances array from updating.